### ISSUE TITLE
SPI Interface Error

### ISSUE THREAD

**Issue by prasuchit** [ArduCAM/Arduino#336]
Hi,

I'm having the exact same error "SPI Interface Error" as https://github.com/ArduCAM/Arduino/issues/141

Please find my circuit and code setup:

![arducam](https://user-images.githubusercontent.com/31422707/41490751-b1565f7c-70b2-11e8-8cb9-768e3dfbce53.JPG)
![img_20180615_153751584_burst000_cover](https://user-images.githubusercontent.com/31422707/41490765-c08d0a2c-70b2-11e8-96bc-3bfc9c4cbbdd.jpg)
![img_20180615_153809206](https://user-images.githubusercontent.com/31422707/41490766-c0b514f4-70b2-11e8-821c-4e220973bc4e.jpg)

Kindly help.

**Comment by prasuchit** [comment]
Thank you @BDCowboy, I shall try that and get back to you asap.

**Comment by UCTRONICS** [comment]
@prasuchit 
Please refer to this schematic diagram 
![icsp_signal_definition](https://user-images.githubusercontent.com/27711713/41579184-72818f56-73c8-11e8-99f9-ee1e42b2e3f2.png)
You should connect the SPI signal to here .

**Comment by prasuchit** [comment]
@UCTRONICS , thanks for the quick response, however, is this the schematic of the arducam board or arduino, I'm confused. The ICSP pin seems to be among those 6 pins at the middle of the arduino mega board, but that wasn't connected in this picture: http://www.arducam.com/wp-content/uploads/2017/12/ArduinoConnection-.jpg

**Comment by prasuchit** [comment]
@UCTRONICS , also I found this link about arduino mega connections, but it is very incoherent, it doesn't mention which sketch it is talking about, doesn't talk about the hardware clearly, and is hard to follow in general: http://www.arducam.com/how-to-connect-arducam-shield-to-mega-2560/
Can you please explain the circuit for arduino mega and any changes in the program? I'm using Arducam_mini_lowpowermode example code from the library.

**Comment by prasuchit** [comment]
Ok, I figured it out, for anyone who is having this trouble in the future: 
(This was done with Arduino Mega, but I guess the principle is the same for all boards)
Connect Vcc and Ground of the arducam board to the +5V and Ground on arduino.  The CS pin from the arducam board to any digital pin on arduino and look for "const int SPI_CS = 49;" in the code that you are running and change that to that pin number. The rest of the pins, refer to the image:
![8c003032-244a-11e7-99a5-eaff19225612](https://user-images.githubusercontent.com/31422707/41612631-b8317268-73b0-11e8-97c5-f468521ebbfd.png)

**Comment by coldeli** [comment]
This worked for me after searching for a really long time. I don't know why arducam wouldn't have a diagram for mega. This worked for me with the 2MP Plus using MEGA

**Comment by UCTRONICS** [comment]
Thanks for your great suggestions. We will update our diagram for mega as soon as possible.

**Comment by ArduCAM** [comment]
Hello,
Arducam has update document and add diagram for mega2560+ Arducam MINI Camera 
https://www.arducam.com/docs/spi-cameras-for-arduino/hardware-connection-for-mega2560/

**Comment by DavidNtlele** [comment]
how do i change the clock speed when using the arduino uno r3 and arducam mini 5mp plus
